integration/securitypolicy_test.go

```diff
@@ -197,16 +294,18 @@ func TestSecurityPolicyMigration(t *testing.T) {
 					copySpec := policyCR.Spec.DeepCopy()
 					copySpec.Name = "Updating Existing Policy"
 
-					err = klient.Universal.Portal().Policy().Update(reqCtx, copySpec)
+					err = updatePolicyOnTyk(reqCtx, copySpec)
 					eval.NoErr(err)
 
 					// Ensure that policy is updated accordingly on Tyk Side.
 					newCopySpec, err := klient.Universal.Portal().Policy().Get(reqCtx, policyCR.Status.PolID)
-					eval.NoErr(err)
+					if err != nil {
+						return false, err
+					}
+
 					eval.True(newCopySpec != nil)
-					eval.Equal(newCopySpec.Name, copySpec.Name)
 
-					return true, nil
+					return newCopySpec.Name == copySpec.Name, nil
```

Comment:
I think it was correct to use eval as before for this check.
Because if name doesn't match there is no point in keep on retrying.